implementation/common/src/test/java/io/smallrye/jwt/util/GenerateSecretFromSignatureAlgoTest.java

```diff
@@ -0,0 +1,41 @@
+package io.smallrye.jwt.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import javax.crypto.SecretKey;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+
+@RunWith(Parameterized.class)
+public class GenerateSecretFromSignatureAlgoTest {
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
```

Comment:
I thin we can have a nice internal map like this inside `KeyUtils` - where we can can get the key bits from the map given an enum value, and also we can get another map like "HS256:HMAC", etc for remaining `HS` and also `A...KW` algos (`AES`) scheme